extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnChunkReaderImpl.java

```diff
@@ -217,12 +217,13 @@ private Dictionary readDictionary(InputStream in) throws IOException {
             // Sometimes the size is explicitly empty, just use an empty payload
             payload = BytesInput.empty();
         } else {
-            payload = BytesInput
-                    .copy(decompressor.decompress(in, compressedPageSize, pageHeader.getUncompressed_page_size()));
+            payload = decompressor.decompress(in, compressedPageSize, pageHeader.getUncompressed_page_size());
         }
-        final DictionaryPage dictionaryPage = new DictionaryPage(payload, dictHeader.getNum_values(),
-                Encoding.valueOf(dictHeader.getEncoding().name()));
-        return dictionaryPage.getEncoding().initDictionary(path, dictionaryPage);
+        final Encoding encoding = Encoding.valueOf(dictHeader.getEncoding().name());
+        final DictionaryPage dictionaryPage = new DictionaryPage(payload, dictHeader.getNum_values(), encoding);
+        // We are safe here because the Dictionary doesn't hold a reference to payload (and thus implicitly, doesn't
+        // hold a reference to the input stream).
```

Comment:
```suggestion
        // We are safe to not copy the payload because the Dictionary doesn't hold a reference to dictionaryPage or
        // payload and thus doesn't hold a reference to the input stream.
```